mystery-frontend/src/Root.tsx

```diff
@@ -1,20 +1,49 @@
+import React, { useEffect } from "react";
 import { QueryClient, QueryClientProvider } from "@tanstack/react-query";
 import { RouterProvider } from "react-router-dom";
+import { getDefaultStore, useSetAtom } from "jotai";
+import { resetUsernameAtom } from "@/atoms/auth";
 import { ThemeProvider } from "@/components/theme-provider";
 import { Toaster } from "@/components/ui/toaster";
 import router from "@/routes";
+import {
+  handleAxiosUnauthorized,
+  handleSocketAuthErrors,
+  clearAxiosMiddleware,
+  clearSocketMiddleware,
+} from "@/services/middleware";
 
 const queryClient = new QueryClient();
 
 function Root() {
   return (
-    <QueryClientProvider client={queryClient}>
-      <ThemeProvider defaultTheme="dark" storageKey="vite-ui-theme">
-        <RouterProvider router={router} />
-        <Toaster />
-      </ThemeProvider>
-    </QueryClientProvider>
+    <MiddlewareHandler>
+      <QueryClientProvider client={queryClient}>
+        <ThemeProvider defaultTheme="dark" storageKey="vite-ui-theme">
+          <RouterProvider router={router} />
+          <Toaster />
+        </ThemeProvider>
+      </QueryClientProvider>
+    </MiddlewareHandler>
   );
 }
 
+function MiddlewareHandler({ children }: { children: React.ReactNode }) {
```

Comment:
Yup, since it's on top of the providers, it will never re-render. And even if it was lower in the provider tree here, it would re-render but the effect wouldn't fire again since the resetUsername function never changes